news/12215.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fix unnecessary error when using ``--dry-run`` and ``--python-version`` without ``--target``
```

Comment:
I would reword this to _Allow `pip install --dry-run` to use platform and ABI overriding options similar to `--target`_ and make the fragment a feature instead. Categorising this as a bugfix with the current wording seems quite convoluted from a user’s perspective.

Reply:
"it's not a big(fix) it's a FEAUTRE!"

😂